deployability/modules/allocation/allocation.py

```diff
@@ -130,10 +136,59 @@ def __generate_track_file(instance: Instance, provider_name: str,  track_path: P
         if not track_path.parent.exists():
             track_path.parent.mkdir(parents=True, exist_ok=True)
+        ssh_config = instance.ssh_connection_info()
         track = models.TrackOutput(identifier=instance.identifier,
                                     provider=provider_name,
                                     instance_dir=str(instance.path),
-                                    key_path=str(instance.credentials.key_path))
+                                    key_path=str(instance.credentials.key_path),
+                                    host_identifier=str(instance.host_identifier),
+                                    ssh_port=ssh_config.port)
         with open(track_path, 'w') as f:
             yaml.dump(track.model_dump(), f)
+        if Path(str(instance.path) + "/port.txt").exists():
+            Path(str(instance.path) + "/port.txt").unlink()
         logger.info(f"Track file generated at {track_path}")
+
+    @staticmethod
+    def __macos_provider(composite_name: str) -> str:
+        """
+        Returns the provider name for macOS instances.
+
+        Args:
+            composite_name (str): The name of the composite.
+
+        Returns:
+            str: The provider name.
+        """
+        if str(composite_name.split("-")[3]) == 'arm64':
+            client = boto3.client('secretsmanager')
+            server_ip = client.get_secret_value(SecretId='devops_macstadium_m1_jenkins_ip')['SecretString']
+            server_port = 22
+            timeout = 5
+
+            conn_ok = False
+            try:
+                tn = Telnet(server_ip, server_port, timeout)
+                conn_ok = True
+                tn.close()
+            except Exception as e:
+                logger.info('Could not connect to macOS macStadium server: ' + str(e) + '. Using AWS provider.')
+
+            if conn_ok:
+                ssh_password = client.get_secret_value(SecretId='devops_macstadium_m1_jenkins_password')['SecretString']
+                ssh_user = client.get_secret_value(SecretId='devops_macstadium_m1_jenkins_user')['SecretString']
+                cmd = "sudo /usr/local/bin/prlctl list -j"
+                prlctl_output = subprocess.Popen(f"sshpass -p {ssh_password} ssh {ssh_user}@{server_ip} {cmd}", shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE).communicate()[0].decode('utf-8')
+                data_list = json.loads(prlctl_output)
+                uuid_count = 0
+                for item in data_list:
+                    if 'uuid' in item:
+                        uuid_count += 1
+                if uuid_count < 2:
+                    logger.info(f"macStadium server has less than 2 VMs running, using Vagrant provider.")
+                    return 'vagrant'
+                else:
+                    logger.info(f"macStadium server has VMs running, using AWS provider.")
+                    return 'aws'
+        if str(composite_name.split("-")[3]) == 'amd64':
+            return 'aws'
```

Comment:
I'm not sure if this logic should be in the `allocation.py` file, maybe if the provider defined by the user in the input is not correct, just raise an error or warning informing that.
I think the user should decide (for good or for bad) what provider use and if its not OK, he will know that system cannot be allocated with that provider when the error/warning is raised.

Reply:
Done: d12bc841c5ca6029f81b541a0b25517f5e4fd131